### PR TITLE
Allow subprocess to exit before exiting the parent process

### LINKED ISSUE #17: On SIGINT, the parent process exits before children
If, for example, a Docker Compose command is running and the user sends a SIGINT. The parent process will appear to immediately exit while the Docker Compose command is still cleaning up. This doesn't seem correct and leads to weird looking shell behavior.

### PR DESCRIPTION
**Pull request by velovix**
Fixes #17

### CODE REVIEW

File: brainframe/cli/main.py [hunk 1]
```diff
@@ -39,9 +39,12 @@ def main():
     # Exit with a clean error when interrupted
     def on_sigint(_sig, _frame):
         print()
-        if os_utils.current_command is not None:
+        if os_utils.current_command is None:
+            print_utils.fail_translate("portal.interrupted")
+        else:
+            # Let os_utils.run take care of bringing the process down when the current
+            # command is finished
             os_utils.current_command.send_signal(_sig)
```
Comment (BryceBeagle) [inline brainframe/cli/main.py:47]: When would `os_utils.current_command` be `None`?

**Review by apockill — APPROVED** [review]
if it works on your machine, it's a:b::b:roved on my end

File: brainframe/cli/main.py (hunk 1 above)
Comment (velovix) [reply brainframe/cli/main.py:47]: That would be when there's no subprocess command running at that moment.

Comment (BryceBeagle) [reply brainframe/cli/main.py:47]: So if there's a current subprocess, we kill it, but then don't run 
```python
print_utils.fail_translate("portal.interrupted")
```
?

Comment (velovix) [reply brainframe/cli/main.py:47]: True. That's unideal. Let me see what I can do.

Comment (velovix) [reply brainframe/cli/main.py]: Okay, I fixed this and added more guarantees.

File: brainframe/cli/os_utils.py [hunk 2]
```diff
@@ -15,7 +16,55 @@
...
+
+    def send_signal(self, sig: int):
+        """
+        :param sig: The signal to send to the subprocess
+        """
+        with self._lock:
+            if self._process is None:
+                raise RuntimeError(
+                    "Attempt to send a signal when no process is running"
+                )
```
Comment (BryceBeagle) [inline brainframe/cli/os_utils.py:62]:
```suggestion
                message = "Attempted to send a signal when no process is running"
                raise RuntimeError(message)
```

File: brainframe/cli/os_utils.py [hunk 3]
```diff
@@ -103,13 +152,18 @@ def run(
...
-    if cmd.returncode != 0 and exit_on_failure:
-        sys.exit(cmd.returncode)
-    return cmd
+
+    current_command.process = subprocess.Popen(command, *args, **kwargs)
+    current_command.process.wait()
+
+    if current_command.interrupted:
+        # A signal was sent to the command before it finished
+        print_utils.fail_translate("general.interrupted")
```
Comment (BryceBeagle) [inline brainframe/cli/os_utils.py:161]: Why not print this at the end of `_CurrentCommand.send_signal`?

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

**Review by BryceBeagle — APPROVED** [review]

**Pull request merged**